From: Hendrik Brueckner <brueckner@redhat.com>
Date: Fri, 7 May 2010 14:35:54 -0400
Subject: [s390] hypfs: fix high cpu time output
Message-id: <20100507143553.GB2607@redhat.com>
Patchwork-id: 24900
O-Subject: [RHEL5.6 PATCH 1/1] [s390x] hypfs: Print unsigned long long values
	using "%llu" format string
Bugzilla: 589282
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>

Description
-----------
For high CPU times, hypfs reports negative values.  The format
string "%lld" is used for sprintf() and unsigned long long variables.
If the first bit becomes 1 then negative values are printed.

Use "%llu" instead of "%lld".

Bugzilla
--------
BZ 589282
https://bugzilla.redhat.com/show_bug.cgi?id=589282

Upstream status of the patch
----------------------------
The patch is upstream as of kernel version 2.6.32
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=ad2a5d8e0b518f997af126dd737127bdada90a6f

Test status
-----------
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.


diff --git a/arch/s390/hypfs/inode.c b/arch/s390/hypfs/inode.c
index 50b27b9..7552edb 100644
--- a/arch/s390/hypfs/inode.c
+++ b/arch/s390/hypfs/inode.c
@@ -414,7 +414,7 @@ struct dentry *hypfs_create_u64(struct super_block *sb, struct dentry *dir,
 	char tmp[TMP_SIZE];
 	struct dentry *dentry;
 
-	snprintf(tmp, TMP_SIZE, "%lld\n", (unsigned long long int)value);
+	snprintf(tmp, TMP_SIZE, "%llu\n", (unsigned long long int)value);
 	buffer = kstrdup(tmp, GFP_KERNEL);
 	if (!buffer)
 		return ERR_PTR(-ENOMEM);
